packages/xrpl/src/models/transactions/transaction.ts

```diff
@@ -213,7 +213,7 @@ export function validate(transaction: Record<string, unknown>): void {
   if (
     !isEqual(
       decode(encode(tx)),
-      omitBy(tx, (value) => value == null),
+      decode(encode(omitBy(tx, (value) => value == null))),
```

Comment:
I think the point of this line of code before was that it didn't do `decode(encode(...))` - it needs to show the original is equivalent to what you're hoping to get. Maybe the change you wanted was to do this to the 1st value in this comparison, and then leave the 2nd value as `omitBy(tx, (value => value == null))` ?

Reply:
@JST5000 to be honest I think this chunk of code should not be here, but rather a part of a test suite. Doesn't make sense as its intentions are to validate the logic, not the input data.

The way to overcome might be something like this you suggest?

```
  if (
    !isEqual(
      tx,
      omitBy(tx, (value) => value == null),
    )
  ) {
    throw new ValidationError(`Invalid Transaction: ${tx.TransactionType}`)
  }
```

Reply:
I'm also thinking that it may just be worth removing the `decode(encode(tx))` block - I'm not sure what value it adds.